src/nycdb/transform.py

```diff
@@ -92,7 +93,22 @@ def to_csv(file_path_or_generator):
 
 def with_bbl(table):
     for row in table:
-        yield merge(row, {'bbl': bbl(row['borough'], row['block'], row['lot'])})
+        borough_key = 'boro' if 'boro' in row else 'borough'
+        yield merge(row, {'bbl': bbl(row[borough_key], row['block'], row['lot'])})
+
+
+def text_to_date(date_columns, table):
+    """
+    Transforms date columns that were stored as text ("19991231")
+    into an actual datetime object
+    """
+    for row in table:
+        for date_column in date_columns:
+            try:
+                row[date_column] = datetime.strptime(row[date_column], '%Y%m%d')
+            except ValueError:
+                row[date_column] = datetime.min
```

Comment:
some fields are malformed so they'll default to the min datetime

Reply:
I think i might be better to return ` None ` here since I'm not sure that storing the date 0001-01-1 in postgres makes much sense. better to be `null`